test/Main.hs

```diff
@@ -48,29 +38,55 @@ dotTest :: String
          -> TestTree
 dotTest testname func genInc = testProperty testname $
     -- Choose the length of the vector
-    forAll (choose (1,10)) $ \ n ->
+    forAll (choose (1,100)) $ \ n ->
     -- Randomly generate two vectors of the chosen length
-    forAll (genNVector (genFloat genEveryday) n) $ \ u ->
-    forAll (genNVector (genFloat genEveryday) n) $ \ v ->
     forAll (genInc) $ \ incx ->
     forAll (genInc) $ \ incy ->
+    forAll (genNVector (genFloat genEveryday) (1+(n-1)*abs incx )) $ \ u ->
+    forAll (genNVector (genFloat genEveryday) (1+(n-1)*abs incy )) $ \ v ->
+
        -- monadically marshal the vectors into arrays for use with CBLAS
        ioProperty $
        withArray (V.toList u) $ \ us ->
        withArray (V.toList v) $ \ vs -> do
            -- compute the expected and observed values
-           expected <- OSX.sdot n us incx vs incy
+           expected <- Fortran.sdot n us incx vs incy
            let observed = func n u incx v incy
            runTest expected observed
-      where
-      runTest expected observed =
-          if expected==observed
-                then return True
-                else do
-                    putStrLn ""
-                    putStrLn $ "EXPECTED: " ++ show expected
-                    putStrLn $ "OBSERVED: " ++ show observed
-                    return False
+
+-- | Evidence that the native sdot function is byte equivalent to the CBLAS
+-- implementation.  Vectors of length 1-10 are tested having elements that are
+-- in the range of approximately (epsilon/2,2/epsilon)
+asumTest :: String
+         -> (Int -> V.Vector Float -> Int -> Float)
+         -> Gen Int
+         -> TestTree
+asumTest testname func genInc = testProperty testname $
+    -- Choose the length of the vector
+    forAll (choose (1,100)) $ \ n ->
+    -- Randomly generate two vectors of the chosen length
+    forAll (genInc) $ \ incx ->
+    forAll (genNVector (genFloat genEveryday) (1+(n-1)*abs incx )) $ \ u ->
+    -- monadically marshal the vectors into arrays for use with CBLAS
+       ioProperty $
+       withArray (V.toList u) $ \ us -> do
+           -- compute the expected and observed values
+           expected <- Fortran.sasum n us incx
+           let observed = func n u incx
+           runTest expected observed
+
+-- | Compare the observed and expected values and print ruman readable error
+-- message.
+runTest :: (Show a, Eq a) => a -> a -> IO Bool
+runTest expected observed =
+  if expected==observed
+        then return True
+        else do
+            putStrLn ""
```

Comment:
The test framework already displays the name of the test and the inputs used.   It just didn't display the outputs.  the runTest function was added to provide that missing information.